File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
    const seen = {};
    
    for (const num of nums) {
        if (seen[num]) {
            return true;
        }
        
        seen[num] = 1;
    }
    
    return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        HashSet<Integer> set = new HashSet<Integer>();

        for (int num: nums) {
            if (set.contains(num)) {
                return true;
            }
            set.add(num);
        }

        return false;
    }
}